### PR TITLE
Add sesv2 data sources for email identity and mail from attributes

### PR DESCRIPTION
**Pull request by jchorl**
### Description
Adds two data sources:
- `aws_sesv2_email_identity`
- `aws_sesv2_email_identity_mail_from_attributes`

### Relations
Relates #26796

### References
I mostly cargo-culted the other two data sources that were present

### Output from Acceptance Testing
I first tested manually:
```
data "aws_sesv2_email_identity" "example" {
  email_identity = "example.com"
}

data "aws_sesv2_email_identity_mail_from_attributes" "example" {
  email_identity = data.aws_sesv2_email_identity.example.email_identity
}

output "tokens" {
  value = data.aws_sesv2_email_identity.example.dkim_signing_attributes[0].tokens
}

output "mail_from" {
  value = data.aws_sesv2_email_identity_mail_from_attributes.example.mail_from_domain
}
```

`terraform refresh` showed the values!

Then I did some automated testing:

```
root@ca07733e14b8:/work# make testacc TESTS=TestAccSESV2EmailIdentityDataSource_basic PKG=sesv2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sesv2/... -v -count 1 -parallel 20 -run='TestAccSESV2EmailIdentityDataSource_basic'  -timeout 180m
=== RUN   TestAccSESV2EmailIdentityDataSource_basic
=== PAUSE TestAccSESV2EmailIdentityDataSource_basic
=== CONT  TestAccSESV2EmailIdentityDataSource_basic
--- PASS: TestAccSESV2EmailIdentityDataSource_basic (48.90s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sesv2      49.039s


root@ca07733e14b8:/work# make testacc TESTS=TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic PKG=sesv2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sesv2/... -v -count 1 -parallel 20 -run='TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic'  -timeout 180m
=== RUN   TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
=== PAUSE TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
=== CONT  TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
--- PASS: TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic (48.79s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sesv2      48.954s
```

### CODE REVIEW

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
$ make testacc PKG=sesv2 TESTS=TestAccSESV2EmailIdentityDataSource_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sesv2/... -v -count 1 -parallel 20 -run='TestAccSESV2EmailIdentityDataSource_'  -timeout 180m
=== RUN   TestAccSESV2EmailIdentityDataSource_basic
=== PAUSE TestAccSESV2EmailIdentityDataSource_basic
=== CONT  TestAccSESV2EmailIdentityDataSource_basic
--- PASS: TestAccSESV2EmailIdentityDataSource_basic (17.86s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sesv2      21.028s
```

```console
$ make testacc PKG=sesv2 TESTS=TestAccSESV2EmailIdentityMailFromAttributesDataSource_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sesv2/... -v -count 1 -parallel 20 -run='TestAccSESV2EmailIdentityMailFromAttributesDataSource_'  -timeout 180m
=== RUN   TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
=== PAUSE TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
=== CONT  TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic
--- PASS: TestAccSESV2EmailIdentityMailFromAttributesDataSource_basic (18.37s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sesv2      21.583s
```

**Comment by jar-b** [comment]
Thanks for your contribution, @jchorl! 👍

File: internal/service/sesv2/email_identity_data_source.go [hunk 1]
```diff
@@ -120,5 +121,5 @@ func dataSourceEmailIdentityRead(ctx context.Context, d *schema.ResourceData, me
 	d.Set("identity_type", string(out.IdentityType))
 	d.Set("verified_for_sending_status", out.VerifiedForSendingStatus)
 
-	return nil
+	return diags
```
Comment (jchorl) [inline internal/service/sesv2/email_identity_data_source.go:124]: For my edification, why is this commit necessary? Do folks add diagnostics in non-error cases? Is this just setting up for a future where we may add diagnostics all throughout this function?

**Comment by jchorl** [comment]
I think this was my first aws tf contribution, so thanks a bunch for cleaning it up and driving it through.

File: internal/service/sesv2/email_identity_data_source.go (hunk 1 above)
Comment (jar-b) [reply internal/service/sesv2/email_identity_data_source.go:124]: The latter - future-proofing for cases where we might add non-error diagnostics. The newer Plugin-Framework [uses this diagnostics pattern](https://developer.hashicorp.com/terraform/plugin/framework/diagnostics), so we try to make Plugin-SDK based resources look similar. 

There are a bunch of legacy Plugin-SDK based resources that don't look this way, but when new stuff comes in its a simple enough change we usually switch over.

**Pull request merged**